### ISSUE TITLE
Running the emulator often hangs when run without "--no-snapshot"

### ISSUE THREAD

**Issue by donv** [ruboto/ruboto#747]
When running 

    ruboto emulator

it always works the first time.  On subsequent times, the emulator most often freezes for me.  When I run with the ```--no-snapshot``` option, it works every time.

There must be something we do wrong when setting the configuration for the emulator that makes snapshots freeze.  (or there may be a bug in the Android emulator/tooling)

**Comment by gfowley** [comment]
I think the problem is with the Android emulator, not the Ruboto emulator command. I have tried many AVD device/configuration combinations and have had similar issues with emulator snapshots. They seem to be a regular source of trouble for Android SDK users:

https://code.google.com/p/android/issues/list?q=emulator+snapshot

This combo of not starting and freezing was the last straw for me:

https://code.google.com/p/android/issues/detail?id=182435

Just a couple of days ago I tried an x86 based AVD for the first time. Wow!.. it zooms.

The x86 emulator will use QEMU+KVM for x86 CPU acceleration and OpenGL GPU acceleration. I am not using snapshots because of the issue above, but with only ~20 seconds from command to working home I don't have a need for them.

I had no trouble getting CPU and GPU acceleration to work - OpenGL and QEMU are just an ```apt-get install``` away on Linux (Ubuntu 14.04). Experiences on OS-X or Windows may be more difficult.

Until now, the ARM emulator performance and snapshot problems caused me to use an actual device for development despite the long ```rake clean debug reinstall``` cycle and ```rake update_scripts``` not working. I will try the x86 AVD for a while and see how it works out.
